projects/components/src/button/button.component.test.ts

```diff
@@ -12,6 +14,7 @@ describe('Button Component', () => {
   const createHost = createHostFactory({
     component: ButtonComponent,
     imports: [ButtonModule, RouterTestingModule],
+    declarations: [MockDirective(TrackDirective)],
```

Comment:
Likewise, should be shallow. So deep tests are forced to mock it, but shallow tests (which all of them need to eventually be migrated to) won't require any changes.

Reply:
true. But that would be a bigger change :(

Reply:
yeah, not asking to change it here, but if it's just a question of adding a mock for the small (hopefully, but maybe not?) number of remaining deep tests, seems better. If that's a big scope increase, can do that later and keep it optional.

Reply:
okay. This rework had come up in the past too. Let's take this as a separate effort. I will keep it optional for now.